pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -440,7 +440,7 @@ func (k *Bootstrapper) StartCluster(cfg config.ClusterConfig) error {
 
 // tunnelToAPIServer creates ssh tunnel between apiserver:port inside control-plane node and host on port 8443.
 func (k *Bootstrapper) tunnelToAPIServer(cfg config.ClusterConfig) error {
-	if cfg.APIServerPort != 0 {
+	if cfg.APIServerPort == 0 {
```

Comment:
how is this not broken for everything else ? or since forever ? is this logic new and only affecting qemu builtin ?

Reply:
Yeah, I was looking into this because our functional tests are failing with this change.

From what I saw this is what happened:

**Before HA PR:**
`tunnelToAPIServer` was only called when it was necessary, ie. QEMU builtin
If it wasn't necessary (docker driver) it wasn't called

**After HA PR but before this fix:**
`tunnelToAPIServer` is getting called for all drivers, but because the logic was broken, it wasn't getting called for any drivers (which broke QEMU w/ builtin)

**This PR**
`tunnelToAPIServer` is getting called for all drivers, and now with the logic fixed it's actually creating a tunnel for all drivers, which fixes QEMU w/ builtin, but Docker (and maybe others) are having issues due to the tunnel because created when it shouldn't be.

So I need to figure out how `tunnelToAPIServer` was getting called prior to the HA change and match it so it's only getting called when necessary.

Reply:
Ah, I found it, before the HA changes `APIServerPort` was only used by QEMU:
```
          if driver.IsQEMU(host.Driver.DriverName()) && network.IsBuiltinQEMU(cfg.Network) {
                  apiServerPort, err := getPort()
                  if err != nil {
                          return runner, preExists, m, host, errors.Wrap(err, "Failed to find apiserver port")
                  }
                  cfg.APIServerPort = apiServerPort
          }
```

But now it's also setting it via a flag:
```
APIServerPort:           viper.GetInt(apiServerPort),
```

The very first test in our functional tests sets the `--apiserver-port` flag, which is starting the tunnel and causing issues.

Reply:
Added the QEMU check below

Reply:
excellent thanks for the investigation